plugins/modules/storagebox_subaccount.py

```diff
@@ -0,0 +1,473 @@
+        default: false
+        required: false
+
+      webdav:
+        description:
+          - Enable or disable WebDAV.
+        type: bool
+        default: false
+        required: false
+
+      readonly:
+        description:
+          - Enable or disable read-only mode.
+        type: bool
+        default: false
+        required: false
+
+      comment:
+        description:
+          - A custom comment for the subaccount.
+        type: str
+        default: ""
+        required: false
+      idempotence:
+        description:
+          - Select which attribute to use to check subaccount existence.
+          - C(username) or C(comment).
+        type: str
+        choices: [username, comment]
+        default: username
+        required: false
+
+notes:
+  - The module is fully idempotent and supports check mode.
+  - Passwords cannot be compared or diffed, so changes to passwords will always appear as changed in check mode and will not show diffs.
+  - When passwords are autogenerated by the API (by omitting the C(password) field), the resulting password is returned.
+"""
+
+EXAMPLES = r"""
+---
+- name: Create a new subaccount with random password
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      homedirectory: "/backups/project1"
+      samba: true
+      ssh: true
+      webdav: false
+      comment: "Backup for Project 1"
+
+- name: Create a subaccount with custom password
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      username: "backup1"
+      password: "s3cretPass123"
+      homedirectory: "/data/backup1"
+      readonly: false
+      samba: true
+      ssh: false
+
+- name: Update an existing subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: present
+      username: "backup1"
+      homedirectory: "/data/backup1-updated"
+      readonly: true
+      comment: "Updated path and readonly mode"
+
+- name: Delete a subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: absent
+      username: "backup1"
+
+- name: Change password for a subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: present
+      username: "backup1"
+      password: "n3wSecur3Pass"
+"""
+
+RETURN = r"""
+created:
+  description: Whether a new subaccount was created.
+  type: bool
+  returned: always
+
+deleted:
+  description: Whether the subaccount was deleted.
+  type: bool
+  returned: always
+
+updated:
+  description: Whether the subaccount's configuration was updated (excluding password changes).
+  type: bool
+  returned: always
+
+password_updated:
+  description: Whether the subaccount's password was updated.
+  type: bool
+  returned: always
+
+created_subaccount:
+  description: The created subaccount object returned from the API.
+  type: raw
+  returned: when created is true
```

Comment:
Why not simply always return the subaccount data?
```suggestion
subaccount:
  description: The subaccount object returned by the API.
  type: dict
  returned: if O(state=present)
```
I'd also add the `contains:` from `storagebox_subaccount_info.py`'s return value docs.

Reply:
I really like the idea of returning Hetzner's response. 
But the "always" part might be a bit more difficult:

- subaccounts updates never returns anything
<img width="1616" alt="image" src="https://github.com/user-attachments/assets/01531319-3b80-4757-bf54-e62facc95c6e" />
so we would be completely fabricating the object

- password updates only return the password
<img width="1303" alt="image" src="https://github.com/user-attachments/assets/626fc48a-0357-4e9b-833a-3efff02c8ed8" />

Return raw created account when creating one, for updates/password updates make a new GET request and append the password if provided ?


P.S: the rest of the review comments should already be fixed